tests/test_agents.py

```diff
@@ -161,10 +161,11 @@ async def crashing_aadd(*args, **kwargs) -> str | None:
             ) as mock_aadd,
         ):
             index = await get_directory_index(settings=agent_test_settings)
+
+    assert len(await index.index_files) == num_source_files
     assert (
-        mock_aadd.await_count <= crash_threshold
+        mock_aadd.await_count != num_source_files
     ), "Should have been able to resume build"
```

Comment:
Probably the most controversial thing on this PR.
As we commit in batchs of 2, we may be building some documents again, but I change the test to see if

1. We end up indexing all docs
2. We don't reindex the whole thing

It's weaker than before, but I think it should be enough